api/spec/v1.yaml

```
@@ -1614,6 +1675,76 @@ components:
       description: |
         A ParaTime block.
 
+    RuntimeEventList:
+      type: object
+      required: [events]
+      properties:
+        events:
+          type: array
+          items:
+            $ref: '#/components/schemas/RuntimeEvent'
+      description: |
+        A list of runtime events.
+
+    RuntimeEvent:
+      type: object
+      required: [round, type, body]
+      properties:
+        round:
+          type: integer
+          format: int64
+          description: The block height at which this event was generated.
+          example: *runtime_block_round_1
+        tx_hash:
+          type: string
+          nullable: true
+          description: |
+            Hash of this event's originating transaction
+            Absent if the event did not originate from a transaction.
+          example: *tx_hash_1
+        type: 
+          description: The type of the event.
+          $ref: '#/components/schemas/RuntimeEventType'
+        body:
+          type: object
+          description: |
+            The decoded event contents. This spec does not encode the many possible types;
+            instead, see [the Go API](https://pkg.go.dev/github.com/oasisprotocol/oasis-sdk/client-sdk/go/modules).
+            This object will conform to one of the `*Event` types two levels down
+            the hierarchy, e.g. `MintEvent` from `Event > accounts.Event > MintEvent`,
+            OR one of the known runtime event types enumerated in EvmRuntimeEvent.
+      description: An event emitted by the runtime layer
+
+    RuntimeEventType:
+      type: string
+      enum:
+        - accounts.transfer
+        - accounts.burn
+        - accounts.mint
+        - consensus_accounts.deposit
+        - consensus_accounts.withdraw
+        - core.gas_used
+        - evm.erc20_transfer
```

```suggestion
        - evm.log
```

At the paratime layer, all EVM events are of a single type: evm.Log.
The SDK [shows this](https://github.com/oasisprotocol/oasis-sdk/blob/b37e6da699df331f5a2ac62793f8be099c68469c/runtime-sdk/modules/evm/src/lib.rs#L302-L309), but then the Go library for interfacing with the SDK just hides away this only subtype, and instead makes `Error` directly contain the three fields that (in Rust) are in Error::Log.

I think it will be confusing if we mix the two kinds (paratime and evm) of event types here. So I'd stick with  just `evm.Log` here.

---

Hm would it be better to flatten/mix the paratime and evm event types? Assuming we're designing for ethereum developers, it would be nice to expose the events in recognizable form, eg `ERC20.transfer`. 

Alternatively, per your suggestion below we could intead return the `evm.Log` raw event, as well as the `evm_log_name` and `evm_log_params` and then the frontend could display it that way. Open to this as well.

---

yes, let's do the latter please, as discussed on the call and in the thread above. We've stuck pretty close to the raw on-chain formats and presentation in other parts of the API, I think it's a good high-level guiding principle.